include/alpaka/kernel/Traits.hpp

```diff
@@ -241,8 +242,8 @@ namespace alpaka
             "The idx type of TAcc and the idx type of TWorkDiv have to be identical!");
 
 #if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
-        std::cout << __func__ << " workDiv: " << workDiv << ", kernelFnObj: " << typeid(kernelFnObj).name()
-                  << std::endl;
+        std::cout << __func__ << " workDiv: " << workDiv
+                  << ", kernelFnObj: " << core::demangled<kernelFnObj> << std::endl;
```

Comment:
You need to pass a type here:
```suggestion
                  << ", kernelFnObj: " << core::demangled<decltype(kernelFnObj)> << std::endl;
```

Reply:
Yeah it's kinda strange, as gcc was complaining in some other place (on my machine) but not there. Whatever :)